File: easy/14. Longest Common Prefix/Solution.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length < 1) return "";

        int minLength = Integer.MAX_VALUE;
        for (String s : strs) {
            if(s.isEmpty()) return "";
            minLength = Math.min(s.length(), minLength);
        }

        for(int i = 0; i < minLength; i++) {
            char c = strs[0].charAt(i);
            for (int j = 1; j < strs.length; j++) {
                if(strs[j].charAt(i) != c) return strs[j].substring(0, i);
            }
        }


        return strs[0].substring(0, minLength);
    }
}

File: easy/14. Longest Common Prefix/Solution.kt
class Solution {
    fun longestCommonPrefix(strs: Array<String>): String {
        if(strs.isEmpty()) return ""

        var min = Int.MAX_VALUE
        strs.forEach { s -> min = min.coerceAtMost(s.length) }

        for(i in 0 until min) {
            val c = strs[0][i]
            for(j in 1 until strs.size) {
                if(strs[j][i] != c) return strs[0].substring(0, i)
            }
        }

        return strs[0].substring(0, min)
    }
}